src/realm/geospatial.cpp

```diff
@@ -271,6 +271,27 @@ std::ostream& operator<<(std::ostream& ostr, const Geospatial& geo)
 
 // The following validation follows the server:
 // https://github.com/mongodb/mongo/blob/053ff9f355555cddddf3a476ffa9ddf899b1657d/src/mongo/db/geo/geoparser.cpp#L140
+
+
+// Technically lat/long bounds, not really tied to earth radius.
+static bool is_valid_lat_lng(double lng, double lat)
+{
+    return abs(lng) <= 180 && abs(lat) <= 90;
+}
+
+static Status coord_to_point(double lng, double lat, S2Point* out)
+{
+    if (!is_valid_lat_lng(lng, lat))
+        return Status(ErrorCodes::InvalidQueryArg,
+                      util::format("Longitude/latitude is out of bounds, lng: %1 lat: %2", lng, lat));
+    // Note that it's (lat, lng) for S2 but (lng, lat) for MongoDB.
+    S2LatLng ll = S2LatLng::FromDegrees(lat, lng).Normalized();
+    // This shouldn't happen since we should only have valid lng/lats.
+    REALM_ASSERT_EX(ll.is_valid(), util::format("coords invalid after normalization, lng = %1, lat = %2", lng, lat));
+    *out = ll.ToPoint();
```

Comment:
This is probably minor, but there is the possibility that the ptr passed here could be null (if we make mistakes in the code), could you pass `S2Point`  by reference?

Reply:
Done!